Markdown editor

amitjha21 opened the pull request:
Changes for markdown editor and preview.  
**Files added/changed:**

- common/vendor/MarkdownEditor/styles/bootstrap-markdown.min.css 
- common/vendor/MarkdownEditor/styles/github.min.css 
- common/vendor/MarkdownEditor/styles/angular-markdown-editor.min.css
- common/vendor/MarkdownEditor/bootstrap-markdown.js 
- common/vendor/MarkdownEditor/highlight.min.js 
- common/vendor/MarkdownEditor/angular-highlightjs.min.js 
- common/vendor/MarkdownEditor/angular-markdown-editor.js 
- Makefile: To add dependency
- recordedit/index.html.in
- test/e2e/utils/recordedithelpers.js

RFSH commented:
@amitjha21 I'm a little confused about the plugins that you have used. Why there are so many different plugins for markdown? Does this rely on marked? I see that for your preview function you're using `renderMarkdown` which is the right markdown renderer.
What's the purpose of each of these plugins? and I don't think we will need french translation.

hongsudt commented:
We should certainly minimize external dependencies as much as we can, so
the app won't be bloated..

On Thu, Jun 22, 2017 at 5:59 PM, Aref Shafaei <notifications@github.com>
wrote:

> @amitjha21 <https://github.com/amitjha21> I'm a little confused about the
> plugins that you have used. Why there are so many different plugins for
> markdown? Does this rely on marked? I see that for your preview function
> you're using renderMarkdown which is the right markdown renderer.
> What's the purpose of each of these plugins? and I don't think we will
> need french translation.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/informatics-isi-edu/chaise/pull/1171#issuecomment-310541754>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAZUDGoPbA1TMMyidjbfFnC8Khwnhqk6ks5sGw4OgaJpZM4OC2_w>
> .[image: Web Bug from
> https://github.com/notifications/beacon/AAZUDNm8Z7UZR1qY_7639Y2Ji3jaK4Y9ks5sGw4OgaJpZM4OC2_w.gif]
> {"api_version":"1.0","publisher":{"api_key":"
> 05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":
> {"external_key":"github/informatics-isi-edu/chaise","
> title":"informatics-isi-edu/chaise","subtitle":"GitHub
> repository","main_image_url":"https://cloud.githubusercontent.com/assets/
> 143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","
> avatar_image_url":"https://cloud.githubusercontent.com/
> assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png
> ","action":{"name":"Open in GitHub","url":"https://github.
> com/informatics-isi-edu/chaise"}},"updates":{"snippets":[{"
> icon":"PERSON","message":"@RFSH in #1171: @amitjha21 I'm a little
> confused about the plugins that you have used. Why there are so many
> different plugins for markdown? Does this rely on marked? I see that for
> your preview function you're using `renderMarkdown` which is the right
> markdown renderer.\r\nWhat's the purpose of each of these plugins? and I
> don't think we will need french translation."}],"action":{"name":"View
> Pull Request","url":"https://github.com/informatics-isi-
> edu/chaise/pull/1171#issuecomment-310541754"}}}

amitjha21 commented:
Removed extra dependency.

RFSH commented:
@amitjha21 The `make all` returns an error. I think you were working on `rowResize.js` and included it in makefile but not the actual file.

jrchudy commented:
If possible, can you use the minified version of all the dependencies you are adding.

amitjha21 commented:
@RFSH Looks like my `git checkout branch` merged my Makefile changes instead of ignoring it. I just removed that line from `Makefile`. It should be good now.

amitjha21 commented:
Added minified versions of dependency which have not been changed.

RFSH commented on common/vendor/MarkdownEditor/angular-markdown-editor.js:
```diff
@@ -0,0 +1,200 @@
+angular
```
Is this directive something that you wrote yourself? Or is it a third-party plugin? If it's thirdparty include its source. If it's something that you wrote or modified add some comments and documentation on how it works.

amitjha21 replied:
It is a third party code that I have modified to suit our requirements. I will write header description for the directive.

amitjha21 replied:
Description for the directive has been added. Commit 40c91da906c89793b6dfb30d25781a1b21523bc1

RFSH commented on common/vendor/MarkdownEditor/angular-markdown-editor.js:
```diff
@@ -0,0 +1,200 @@
+
+function addNewButtons() {
+  return [[
+        // {
```
If these are unnecessary, just remove them instead of commenting.

amitjha21 replied:
These are buttons which are not included right now but might be useful to include in future. I left it commented for future enhancements.

RFSH commented on test/e2e/utils/recordedit-helpers.js:
```diff
@@ -317,62 +321,78 @@ exports.testPresentationAndBasicValidation = function(tableParams, isEditMode) {
+                                modalPrevBtn.click();
+                                let mdDiv = element(by.css('[ng-bind-html="ctrl.params.markdownOut"]'));
+                                browser.wait(EC.presenceOf(mdDiv), browser.params.defaultTimeout);
+                                expect(mdDiv.getAttribute('outerHTML')).toContain(markdownOut, "Error during markdown pop-up generation");
```
You can wrap these error messages in the helper function that I wrote in this function to make it easier to track where it exactly failed. `colError(descCol.name, "Error during markdown pop-up generation")`. It will just provide the column name and row index that this error occurred in it.

amitjha21 replied:
Added. 40c91da906c89793b6dfb30d25781a1b21523bc1

RFSH commented on test/e2e/utils/recordedit-helpers.js:
```diff
@@ -317,62 +321,78 @@ exports.testPresentationAndBasicValidation = function(tableParams, isEditMode) {
+                                PrevBtn.click();        //generate preview
+                                let mdPrevDiv = element(by.className("md-preview"));
+                                browser.wait(EC.presenceOf(mdPrevDiv), browser.params.defaultTimeout);
+                                expect(mdPrevDiv.getAttribute('outerHTML')).toContain(markdownOut, "Error during markdown preview generation");
```
Same thing about this error.

amitjha21 replied:
Added. 40c91da

jrchudy commented on test/e2e/utils/recordedit-helpers.js:
```diff
@@ -7,40 +7,44 @@ var EC = protractor.ExpectedConditions;
 
 //test params for markdownPreview
 var markdownTestParams = [{
```
Are the markdown test params used for both edit and create?

amitjha21 replied:
Yes, the module `testPresentationAndBasicValidation` is called from both **add** and **edit** spec files in _recordedit_. This module contains most of the code in `recordedit-helpers.js` including  that for markdown test.

amitjha21 commented:
Added changes for test case to accommodate Protractor version issue with **Select ALL(CTRL+A)** key command. Refactored code does not use the keyboard command anymore. 
`innerHTML` added.
 40c9a6be12c4f4ab91e1a5a2d404a598ba6dd7e5
